test/boost/secondary_index_test.cc

```diff
@@ -1560,6 +1560,305 @@ SEASTAR_TEST_CASE(test_token_order) {
+
+            msg = cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND token(pk) >= {} AND token(pk) <= {} ALLOW FILTERING", testset_tokens[1], testset_tokens[4] - 1).c_str());
+            assert_that(msg).is_rows().with_rows(get_result_rows(1, 3));
+
+            msg = cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND token(pk) >= {} AND token(pk) <= {} ALLOW FILTERING", testset_tokens[1], testset_tokens[4] + 1).c_str());
+            assert_that(msg).is_rows().with_rows(get_result_rows(1, 4));
+
+            msg = cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND token(pk) >= {} AND token(pk) <= {} ALLOW FILTERING", testset_tokens[1] + 1, testset_tokens[4]).c_str());
+            assert_that(msg).is_rows().with_rows(get_result_rows(2, 4));
+
+            // exclusive-inclusive range
+            msg = cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND token(pk) > {} AND token(pk) <= {} ALLOW FILTERING", testset_tokens[1], testset_tokens[4]).c_str());
+            assert_that(msg).is_rows().with_rows(get_result_rows(2, 4));
+
+            // inclusive-exclusive range
+            msg = cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND token(pk) >= {} AND token(pk) < {} ALLOW FILTERING", testset_tokens[1], testset_tokens[4]).c_str());
+            assert_that(msg).is_rows().with_rows(get_result_rows(1, 3));
+
+            // exclusive-exclusive range
+            msg = cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND token(pk) > {} AND token(pk) < {} ALLOW FILTERING", testset_tokens[1], testset_tokens[4]).c_str());
+            assert_that(msg).is_rows().with_rows(get_result_rows(2, 3));
+
+            // inclusive-infinity range
+            msg = cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND token(pk) >= {} ALLOW FILTERING", testset_tokens[3]).c_str());
+            assert_that(msg).is_rows().with_rows(get_result_rows(3, testset_pks.size() - 1));
+
+            // exclusive-infinity range
+            msg = cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND token(pk) > {} ALLOW FILTERING", testset_tokens[3]).c_str());
+            assert_that(msg).is_rows().with_rows(get_result_rows(4, testset_pks.size() - 1));
+
+            // infinity-inclusive range
+            msg = cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND token(pk) <= {} ALLOW FILTERING", testset_tokens[3]).c_str());
+            assert_that(msg).is_rows().with_rows(get_result_rows(0, 3));
+
+            // infinity-exclusive range
+            msg = cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND token(pk) < {} ALLOW FILTERING", testset_tokens[3]).c_str());
+            assert_that(msg).is_rows().with_rows(get_result_rows(0, 2));
+
+            // negative-positive range
+            auto start_index =  3;
+            auto end_index = 5;
+            assert(testset_tokens[start_index] < 0);
+            assert(testset_tokens[end_index] > 0);
```

Comment:
Nitpick: why does it even matter if the token is < 0 or > 0? are you trying to reproduce some potential bug involving encoding of negative tokens or something? Did we have such bug?

Reply:
It's leftover from the time when I implemented this PR for older indexes (those with `bytes` `idx_token` column) and this triggered different code path (due to `bytes` being unsigned).

Reply:
In the amended commit I deleted this query (but slightly modified some other query to also be over negative-positive range, but without `assert` check).